packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -523,26 +529,49 @@ export class SwapQuoter {
             gasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
         }
         // get the relevant orders for the makerAsset
-        let prunedOrders = await this._getSignedOrdersAsync(makerAssetData, takerAssetData);
+        const orderPromises: Array<Promise<SignedOrder[]>> = [
+            this._getSignedOrdersAsync(makerAssetData, takerAssetData),
+        ];
+        if (
+            options.rfqt &&
+            options.rfqt.intentOnFilling &&
+            options.apiKey &&
+            this._rfqtTakerApiKeyWhitelist.includes(options.apiKey)
+        ) {
+            if (!options.rfqt.takerAddress || options.rfqt.takerAddress === constants.NULL_ADDRESS) {
+                throw new Error('RFQ-T requests must specify a taker address');
+            }
+            orderPromises.push(
+                this._quoteRequestor.requestRfqtFirmQuotesAsync(
+                    makerAssetData,
+                    takerAssetData,
+                    assetFillAmount,
+                    marketOperation,
+                    options.apiKey,
+                    options.rfqt.takerAddress,
+                ),
+            );
+        }
+        const orders: SignedOrder[] = ([] as SignedOrder[]).concat(...(await Promise.all(orderPromises)));
```

Comment:
Curious as to what's up with the craziness in the line?  Can we just do the following?

```
const orders: SignedOrder[] = await Promise.all(orderPromises);
```